maven-plugins/helidon-archetype-maven-plugin/src/main/java/io/helidon/build/maven/archetype/JarMojo.java

```diff
@@ -343,7 +465,13 @@ private void processArchetypeResources(Map<String, List<String>> resources,
                     printer.println(resource);
                     Path resourceTarget = archetypeDir.resolve(resource);
                     getLog().debug("adding resource to archetype directory: " + resource);
-                    Files.createDirectories(resourceTarget);
+                    if (resourceTarget.toFile().isDirectory()) {
+                        Files.createDirectories(resourceTarget);
```

Comment:
This check should not be required as `Files.createDirectories` creates the non existing parents.

> Creates a directory by creating all nonexistent parent directories first[...]

Reply:
without this check the build fails with `java.nio.file.FileAlreadyExistsException`

Reply:
The 2nd sentence of the Javadoc for `createDirectories` states that it shouldn't fail if the file already exists:

>  Unlike the createDirectory method, an exception is not thrown if the directory could not be created because it already exists.

Loo

Reply:
Right, but the exception is thrown when `resourceTarget` is a file (not a directory).

Reply:
Those paths are always files, that's why I'm thinking that the original code was wrong and should have used `Files.createDirectories(resourceTarget.getParent())`

Reply:
Not sure why it's failing now but not with 2.x.

Reply:
It is failing when any file is added in `resources` (not in `META-INF`)